csrc/index_compute.cpp

```diff
@@ -1214,7 +1214,7 @@ void ensureStaticIndexing(
       continue;
     }
     IterDomain* loop_id = loop->iter_domain();
-    if (loop->vectorize() || loop_id->isThread()) {
+    if (loop->vectorize() || loop_id->isThread() || loop_id->isDeviceDim()) {
```

Comment:
Can you please change this line to:

```suggestion
    if (loop->vectorize() || isMemoryPartitionedAcross(tv->getMemoryType, loop_id->getParallelType()) {
```

Basically, what we are doing here is to figure if each loop contributes to the index of the given tensor. If so, the loop needs to be fully unrolled to make the index static. That's important since dynamic indices would prevent the use of registers.

This particular line is just filtering out loops that won't matter. For register tensors, TID/BID parallelized loops won't matter as registers are thread local. That's obviously the case too for DID. 

`isMemoryPartitionedAcross` is a utility function we recently added to make this logic a little easier to change and extend. If we ever add another memory type, this would help reduce code changes.